blog/2019-10-02-MigratingOptaWebEmployeeRosteringToSpringBoot.adoc

```diff
@@ -0,0 +1,204 @@
+
+[source,properties]
+----
+optaweb.generator.initial.data=EMPTY
+----
+
+This sets the `optaweb.generator.initial.data` system property for the application when it starts.
+
+== REST API
+
+Now let's actually interact with the backend application.
+An easy way to test the REST API is to use the Swagger framework, which OptaWeb Employee Rostering supports.
+
+As an example, here are the `Tenant` REST methods shown in the Swagger user interface:
+
+image::tenantMethods.png[]
+
+As we can see, each method has documentation, the relative path URL, and its corresponding HTTP request.
+The Swagger UI makes it very easy to test any method in the backend.
+Let's say we wanted a list of all the Tenants in the application.
+We start by selecting the corresponding method to display the method details:
+
+image::getTenantsMethod.png[]
+
+Here, the `Example Value` field shows a sample value of a Tenant list, serialized to JSON format.
+We see that each Tenant contains the `id`, `name`, and `version` fields, which are saved in the database whenever a Tenant is added.
+Also note the `Try it out` button in the top right corner.
+We can use it to send HTTP requests with ease for testing purposes.
+Let's try it out:
+
+image::executeGetTenants.png[]
+
+Sending the GET request returns all the saved Tenants.
+We can see them in JSON format above.
+
+Nice, we just sent our first HTTP request!
+But the Employee Rostering application does much more than just getting entities.
+There are methods for creating, deleting, and updating various entities (remember the <<Domain Model>> above?), which the application uses to modify the database, as well as methods that interact with the OptaPlanner API.
+
+Let's try a more interesting method:
+
+image::solveMethod.png[]
+
+This method is the core of the OptaWeb Employee Rostering application.
+It starts the OptaPlanner solver that assigns work shifts to employees for a specific Tenant.
+The solver uses heuristics and algorithms to find the most optimal solution under time and resource constraints.
+To learn more about how OptaPlanner finds better solutions, read about https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/index.html#localSearch[Local Search] in the OptaPlanner docs.
+
+Let's try out the solver for the Tenant with `id = 1`:
+
+image::executeSolverMethod.png[]
+
+After clicking `Execute`, the solver starts evaluating possible solutions for arranging shifts, and finds the shift roster with the best score.
+The score is calculated using the Drools business constraint rules that we configured in the application.
+As the algorithm finds new solutions, the engine evaluates the score of each solution and compares it with the score of the current best solution.
+If the score is higher, the new solution becomes the new best solution.
+We can see this process in the messages the logger produces during solving:
```

Comment:
Try to remove all the sentence parts that are just ceremony. Add where those logging message appear (= in the console).
-> "Notice the log messages during solving in the console:"